tests/mpi/refine_and_coarsen_fixed_number_08.mpirun=1.with_p4est=true.output

```diff
@@ -0,0 +1,37 @@
+
+DEAL::n_cells:100
+DEAL::criteria:[b>0,e>0]
+DEAL::  refinement_fraction:0.1 coarsening_fraction:0.1
+DEAL::    n_refine_flags:10 n_coarsen_flags:10
+DEAL::  refinement_fraction:0.1 coarsening_fraction:0.
+DEAL::    n_refine_flags:10 n_coarsen_flags:0
+DEAL::  refinement_fraction:0. coarsening_fraction:0.1
+DEAL::    n_refine_flags:0 n_coarsen_flags:10
+DEAL::criteria:[b=0,e>0]
+DEAL::  refinement_fraction:0.1 coarsening_fraction:0.1
+DEAL::    n_refine_flags:10 n_coarsen_flags:10
+DEAL::  refinement_fraction:0.1 coarsening_fraction:0.
+DEAL::    n_refine_flags:10 n_coarsen_flags:1
```

Comment:
There was a bit more to modify to allow negative criteria, which only make sense with the `fixed_number` strategy by the way.

I provided corresponding changes including a corresponding test which looks good. However, I'm not happy with this single cell being coarsened. I'll investigate.

Reply:
The cause of this were these lines. https://github.com/dealii/dealii/blob/c4c3077c421f758bf76f8eca539a7bdda80415fb/source/distributed/grid_refinement.cc#L549-L551
If no cells will be coarsened and the minimum of all criteria is zero, then the bottom_threshold is also zero and not lower than the minimum. Setting bottom_threshold to std::numeric_limits<Number>::lowest() did the trick